File: HillHigh1980.Core/DomainService/IRosterRepository.cs
using System;
using HillHigh1980.Core.Entity;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace HillHigh1980.Core.DomainService
{
    public interface IRosterRepository
    {
        Task<IEnumerable<Roster>> ReadAll(IFilterData filter = null);
        Task<Roster> FindById(int rosterId);
        Task<IEnumerable<Roster>> FindRostersByName(Filter filter);
        Task<Location> CreateLocation(Location location);
        Task<IEnumerable<Location>> GetLocations(int rosterId);
        Task<Location> UpdateLocation(Location location);
        Task DeleteLocation(Location location);
    }
}

File: HillHigh1980.UI/wwwroot/Script/Core/RepositoryService/IRosterRepository.ts
module GScope {
    export module RepositoryService {
        export interface IRosterRepository {
            ReadAll(): Promise<any>;
            FindById(id: number): Promise<any>;
            FindByName(filter: Entity.Filter):Promise<any>
            CreateLocations(locations: Entity.Location): JQuery.Promise<any>;
            UpdateLocation(location: Entity.Location): Promise<any>;
            DeleteLocation(location: Entity.Location): Promise<any>;
            GetGPS: (location:Entity.Location) => JQuery.Promise<any>;
            PartialViewById: (id: number) => Promise<any>
        }
    }
}